src/components/Header/index.tsx

```diff
@@ -1,16 +1,70 @@
+import { Box, Modal, Typography } from '@mui/material';
 import Image from 'next/image';
+import { useState } from 'react';
 
 import Button from '../Button';
+import TextInput from '../TextInput';
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
 
 export default function Header() {
+  const [isOpenModal, setIsOpenModal] = useState(false);
+  const handleOpen = () => setIsOpenModal(true);
+  const handleClose = () => setIsOpenModal(false);
```

Comment:
For short event handlers (1-2 lines), it is common to put them right at the JSX, as an annonymous arrow function.